src/locale/bn/_lib/localize/index.js

```diff
@@ -90,56 +118,73 @@ var formattingDayPeriodValues = {
   }
 }
 
-function ordinalNumber (dirtyNumber, dirtyOptions) {
-  var number = Number(dirtyNumber)
-
-  // If ordinal numbers depend on context, for example,
-  // if they are different for different grammatical genders,
-  // use `options.unit`:
-  //
-  //   var options = dirtyOptions || {}
-  //   var unit = String(options.unit)
-  //
-  // where `unit` can be 'year', 'quarter', 'month', 'week', 'date', 'dayOfYear',
-  // 'day', 'hour', 'minute', 'second'
-
-  // refer to https://bn.wikipedia.org/wiki/পূরণবাচক_সংখ্যা_(ভাষাতত্ত্ব)
-  var rem100 = number % 100
-  if (rem100 <= 10) {
-    switch (rem100 % 10) {
+function dateOrdinalNumber(number, localeNumber) {
+  if (number > 18 && number <= 31) {
+    return localeNumber + 'শে'
+  }
+  else {
+    switch(number) {
+      case 1:
+        return localeNumber + 'লা'
       case 2:
       case 3:
-        return number + 'য়'
+        return localeNumber + 'রা'
       case 4:
-        return number + 'র্থ'
-      case 6:
-        return number + 'ষ্ঠ'
-      case 1:
-      case 5:
-      case 7:
-      case 8:
-      case 9:
-      case 0:
-        return number + 'ম'
+        return localeNumber + 'ঠা'
+      default:
+      return localeNumber + 'ই'
     }
   }
-  if (rem100 > 48 && rem100 < 59) {
-    return number + 'ৎ'
-  }
-  if (rem100 > 58 && rem100 < 69) {
-    return number + 'ষ্টি'
-  }
-  if (rem100 > 68 && rem100 < 99) {
-    return number + 'তি'
+}
+
+function ordinalNumber (dirtyNumber, dirtyOptions) {
+  var number = localize.localeToNumber(dirtyNumber)
+  var localeNumber = localize.numberToLocale(number)
+  const { unit } = dirtyOptions;
+
+  if (unit === 'date') {
+    return dateOrdinalNumber(number, localeNumber)
   }
-  if (rem100 === 0) {
-    return number + 'ত'
+  if (number > 10 || number === 0) return localeNumber + 'তম'
+
+  var rem10 = number % 10
+  switch (rem10) {
+    case 2:
+    case 3:
+      return localeNumber + 'য়'
+    case 4:
+      return localeNumber + 'র্থ'
+    case 6:
+      return localeNumber + 'ষ্ঠ'
+    case 1:
+    case 5:
+    case 7:
+    case 8:
+    case 9:
+    case 0:
```

Comment:
As  `case 0` is handled before in the `if` block, we don't need it here anymore. Or we can add another case `0 - তম ` and remove it from the `if` block. Please change it to whichever you like.

Reply:
This case 0 is for `10`. 10 % 10 returns 0.

Reply:
Yes, but just before the switch block you already handled `case 0`:

`if (number > 10 || number === 0) return localeNumber + 'তম'`.